src/main.cpp

```diff
@@ -1632,12 +1628,14 @@ bool CheckInputs(const CTransaction& tx, CValidationState &state, const CCoinsVi
             const CCoins *coins = inputs.AccessCoins(prevout.hash);
             assert(coins);
 
-            // If prev is coinbase, check that it's matured
             if (coins->IsCoinBase()) {
-                if (nSpendHeight - coins->nHeight < COINBASE_MATURITY)
+                // Ensure that coinbases cannot be spent to transparent outputs
+                // Disabled on regtest
+                if (consensusParams.coinbaseMustBeProtected && !tx.vout.empty()) {
```

Comment:
Doesn't this still allow a coinbase to be transparently "spent" as miner fees, thus potentially getting around the requirement for protecting coinbases? (at the expense of potentially losing the coins, so maybe it doesn't matter...)